Content.Server/GameObjects/Components/ActionBlocking/CuffedComponent.cs

```diff
@@ -0,0 +1,352 @@
+                    ignoredEnt: Owner))
+            {
+                _notifyManager.PopupMessage(user, user, "You are too far away to remove the cuffs.");
+                return;
+            }
+
+            if (!EntitySystem.Get<SharedInteractionSystem>().InRangeUnobstructed(
+                    cuffsToRemove.Transform.MapPosition,
+                    Owner.Transform.MapPosition,
+                    _interactRange,
+                    ignoredEnt: Owner))
+            {
+                Logger.Warning($"Handcuffs being removed from player are obstructed or too far away! This should not happen!");
+                return;
+            }
+
+            _notifyManager.PopupMessage(user, user, "You start removing the cuffs.");
+            _audioSystem.PlayFromEntity(isOwner ? cuff.StartBreakoutSound : cuff.StartUncuffSound, Owner);
+
+            var uncuffTime = isOwner ? cuff.BreakoutTime : cuff.UncuffTime;
+            var doAfterEventArgs = new DoAfterEventArgs(user, uncuffTime)
+            {
+                BreakOnUserMove = true,
+                BreakOnDamage = true,
+                BreakOnStun = true,
+                NeedHand = true
+            };
+
+            var result = await _doAfterSystem.DoAfter(doAfterEventArgs);
+
+            if (result != DoAfterStatus.Cancelled)
+            {
+                _audioSystem.PlayFromEntity(cuff.EndUncuffSound, Owner);
+
+                _container.ForceRemove(cuffsToRemove);
+                cuffsToRemove.Transform.AttachToGridOrMap();
+                cuffsToRemove.Transform.WorldPosition = Owner.Transform.WorldPosition;
+
+                if (cuff.BreakOnRemove)
+                {
+                    cuff.Broken = true;
+
+                    cuffsToRemove.Name = cuff.BrokenName;
+                    cuffsToRemove.Description = cuff.BrokenDesc;
+
+                    if (cuffsToRemove.TryGetComponent<SpriteComponent>(out var sprite))
+                    {
+                        sprite.LayerSetState(0, cuff.BrokenState); // TODO: safety check to see if RSI contains the state?
+                    }
+                }
+
+                CanStillInteract = _hands.Hands.Count() > CuffedHandCount;
+                OnCuffedStateChanged.Invoke();
+                UpdateStatusEffect();
+                Dirty();
+
+                if (CuffedHandCount == 0)
+                {
+                    _notifyManager.PopupMessage(user, user, Loc.GetString("You successfully remove the cuffs."));
+
+                    if (!isOwner)
+                    {
+                        _notifyManager.PopupMessage(user, Owner, Loc.GetString($"{user.Name} uncuffs your hands."));
+                    }
+                }
+                else
+                {
+                    if (!isOwner)
+                    {
+                        _notifyManager.PopupMessage(user, user, Loc.GetString($"You successfully remove the cuffs. {CuffedHandCount} of {user.Name}'s hands remain cuffed."));
```

Comment:
theName
Also move the CufferHandCount to the arguments.

Loc.GetString will not translate most interpolated strings (because interpolation comes first)